Sources/CacheAdvance/CacheAdvance.swift

```diff
@@ -172,10 +170,28 @@ public final class CacheAdvance<T: Codable> {
         try header.checkFile()
 
         var messages = [T]()
-        while let encodedMessage = try reader.nextEncodedMessage() {
-            messages.append(try decoder.decode(T.self, from: encodedMessage))
-        }
+        // There is only one range: | `offsetInFileOfOldestMessage` -> `offsetInFileAtEndOfNewestMessage`|
+        if reader.offsetInFileOfOldestMessage < reader.offsetInFileAtEndOfNewestMessage {
```

Comment:
When the cache is empty I assume that `reader.offsetInFileOfOldestMessage == reader.offsetInFileAtEndOfNewestMessage`? I wonder if it would be best to explicitly handle the case of these two values being equal, to make this code easier to reason about.

Reply:
It would also be nice to validate my assumption that `reader.offsetInFileOfOldestMessage == reader.offsetInFileAtEndOfNewestMessage` when the cache is empty.

Reply:
Per this comment block
https://github.com/dfed/CacheAdvance/blob/3a1ed6f99c2a870dc09c3a8144a58f915757a569/Tests/CacheAdvanceTests/CacheAdvanceTests.swift#L367-L368

when `reader.offsetInFileOfOldestMessage` == `reader.offsetInFileAtEndOfNewestMessage`, the file is empty. I like the idea of explicitly handling this case.

Reply:
like this idea too. handle `==` as empty will make it more clear